src/Tools/dotnet-counters/CounterMonitor.cs

```diff
@@ -1,37 +1,56 @@
-// Copyright (c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
 
 using Microsoft.Diagnostics.Tools.RuntimeClient;
 using System;
+using System.Collections.Generic;
 using System.CommandLine;
 using System.IO;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 
+using Microsoft.Diagnostics.Tracing;
+
 namespace Microsoft.Diagnostics.Tools.Counters
 {
     public class CounterMonitor
     {
-        private string outputPath;
-        private ulong sessionId;
-
         private int _processId;
         private int _interval;
         private string _counterList;
         private CancellationToken _ct;
         private IConsole _console;
-
+        private ConsoleWriter writer;
+        private ulong _sessionId;
         public CounterMonitor()
         {
+            writer = new ConsoleWriter();
+        }
+
+        private void Dynamic_All(TraceEvent obj)
+        {
+            if (obj.EventName.Equals("EventCounters"))
+            {
+                IDictionary<string, object> payloadVal = (IDictionary<string, object>)(obj.PayloadValue(0));
+                IDictionary<string, object> payloadFields = (IDictionary<string, object>)(payloadVal["Payload"]);
+
+                // There really isn't a great way to tell whether an EventCounter payload is an instance of 
+                // IncrementingCounterPayload or CounterPayload, so here we check the number of fields 
+                // to distinguish the two.                
+                ICounterPayload payload = (payloadFields.Count == 6) ? (ICounterPayload)new IncrementingCounterPayload(payloadFields) : (ICounterPayload)new CounterPayload(payloadFields);
+                
+                writer.Update(obj.ProviderName, payload);
+            }
         }
 
-        public async Task<int> Monitor(CancellationToken ct, string counterList, IConsole console, int processId, int interval)
+        public async Task<int> Monitor(CancellationToken ct, string counter_list, IConsole console, int processId, int interval)
         {
             try
             {
                 _ct = ct;
-                _counterList = counterList;
+                _counterList = counter_list; // NOTE: This variable name has an underscore because that's the "name" that the CLI displays. System.CommandLine doesn't like it if we change the variable to camelcase. 
```

Comment:
This is fine, but if you haven't already give feedback to the System.CommandLine project when you hit these rough edges.

Reply:
Will do!